website/docs/ref/react.doc.js

```diff
@@ -197,6 +197,28 @@ function renderButton(container: HTMLElement, visited?: boolean) {
 
   Any function that returns a React element can be used as a component class in
   a JSX expression.
+
+*/
+const AsyncGreeter = loadAsync(Greeter, new Promise((resolve, reject) => {
```

Comment:
What do you find weird about it?

I would hope the tradeoff isn't drastic between the two that it could be simple and type safe